javascript/app-model-exposure/src/api-client/FetchApiInteractor.utils.ts

```diff
@@ -16,10 +16,11 @@ export async function checkStatus(
 	_updateId?: string,
 ): Promise<StatusResponse> {
 	let status = await api.fetchStatus<StatusResponse>(taskId, type)
-
-	while (isProcessingStatus(status.status as NodeResponseStatus)) {
+	while (
+		isProcessingStatus(status.status as NodeResponseStatus, status.pending)
+	) {
 		// eslint-disable-next-line @essex/adjacent-await
-		await wait(3000)
+		await wait(10000)
```

Comment:
why are we waiting explicitly here? shouldn't the await on line 25 be fine?

Reply:
running it locally I've experienced a freeze in the FE even with a wait of 3secs, that's the reason why I incremented the wait to 10secs

Reply:
Having an arbitrary wait is an indication of some other underlying async problem. Unless we figure out how to wait for exactly what we need to wait for, any amount of time could cause a failure (let's say it takes 12 seconds once every 20 runs...)